airbyte-cdk/python/airbyte_cdk/sources/declarative/manifest_declarative_source.py

```diff
@@ -84,6 +93,27 @@ def spec(self, logger: logging.Logger) -> ConnectorSpecification:
         else:
             return super().spec(logger)
 
+    def check(self, logger: logging.Logger, config: Mapping[str, Any]) -> AirbyteConnectionStatus:
+        self._configure_logger_level(logger)
+        return super().check(logger, config)
+
+    def read(
+        self,
+        logger: logging.Logger,
+        config: Mapping[str, Any],
+        catalog: ConfiguredAirbyteCatalog,
+        state: Union[List[AirbyteStateMessage], MutableMapping[str, Any]] = None,
+    ) -> Iterator[AirbyteMessage]:
+        self._configure_logger_level(logger)
```

Comment:
set log level to debug if debug mode is enabled

Reply:
for the next breaking change, i feel like we should remove the `logger` from these parameters. It should be bound to the instance.

Reply:
Created a CDK v2 wishlist https://github.com/airbytehq/airbyte/issues/19239